src/mesh/unstructuredMesh/BoundaryMesh.cpp

```diff
@@ -0,0 +1,149 @@
+// SPDX-License-Identifier: MIT
+// SPDX-FileCopyrightText: 2023 NeoFOAM authors
+
+#include "NeoFOAM/mesh/unstructuredMesh/BoundaryMesh.hpp"
+
+namespace NeoFOAM
+{
+
+BoundaryMesh::BoundaryMesh(
+    const Executor& exec,
+    labelField faceCells,
+    vectorField cf,
+    vectorField cn,
+    vectorField sf,
+    scalarField magSf,
+    vectorField nf,
+    vectorField delta,
+    scalarField weights,
+    scalarField deltaCoeffs,
+    std::vector<localIdx> offset
+)
+    : exec_(exec),
+      faceCells_(faceCells),
+      Cf_(cf),
+      Cn_(cn),
+      Sf_(sf),
+      magSf_(magSf),
+      nf_(nf),
+      delta_(delta),
+      weights_(weights),
+      deltaCoeffs_(deltaCoeffs),
+      offset_(offset) {};
+
+// Accessor methods
+const labelField& BoundaryMesh::faceCells() const
+{
+    return faceCells_;
+}
+
+std::span<const label> BoundaryMesh::faceCells(const localIdx i) const
+{
+    label start = offset_[i];
+    label end = offset_[i + 1];
```

Comment:
As I understand the below suggestion is considered a pure alais, and so the compiler will actually delete the temporary - I am sure it will also with 03 etc, but if you are in debug mode it can be nice to try to keep some performance (if these sorts of things are done through a code base). If we don't care at this stage/or at all then you can just ignore the comment, otherwise, there are of course further examples below. 

```suggestion
    const label& start = offset_[i];
    const label& end = offset_[i + 1];
```